Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp

```diff
@@ -406,6 +390,60 @@ static void parse_normalization_props(Core::File& file, UnicodeData& unicode_dat
     }
 }
 
+static void add_canonical_code_point_name(CodePointRange range, StringView name, UnicodeData& unicode_data)
+{
+    // https://www.unicode.org/versions/Unicode14.0.0/ch04.pdf#G142981
+    // FIXME: Implement the NR1 rules for Hangul syllables.
+
+    // These code point ranges are the NR2 set of name replacements defined by Table 4-8.
+    constexpr Array<CodePointName, 15> s_ideographic_replacements { {
+        { { 0x3400, 0x4DBF }, "CJK UNIFIED IDEOGRAPH-{:X}"sv },
```

Comment:
Most of these ranges already appear in `UnicodeData.txt`, sometimes with a slightly more informative name. `CJK Ideograph Extension A` seems slightly more informative than just `CJK UNIFIED IDEOGRAPH`.

I agree that manually spoofing the Khitan, Nushu, and `{ 0x2F800, 0x2FA1D }` CJK-ranges are a good idea.

The others? I don't really see the point, and I think using a general `String::formatted("{}{}", name, "-{:X}")` would be a better fit. Less hardcoded data, less breakage if future Unicode versions tweak those ranges.

This would require that the ranges are only replaced on the *last* code point of a range; but that should be easily doable, just adapt the iterator logic below.

Reply:
> CJK Ideograph Extension A seems slightly more informative than just CJK UNIFIED IDEOGRAPH.

That's actually directly against the Unicode standard. The name of code point U+3400 is "CJK UNIFIED IDEOGRAPH-3400". Take a look at [DerivedName.txt](https://www.unicode.org/Public/14.0.0/ucd/extracted/DerivedName.txt), it's a good cross-reference to see the formal names assigned by the standard. (Note we could parse that file, instead of implementing the rules here, but that unintuitively seemed like more work because that file leaves out control / private / etc. code points).

> Less hardcoded data, less breakage if future Unicode versions tweak those ranges.

Ideally the Unicode standard would publish these formatting rules in a UCD file that we could parse, and not hard code them. But they don't, unfortunately. That said, the standard is quite good at broadcasting breaking changes in new Unicode versions, so if they change these ranges, it'd be highlighted.

Reply:
All that said, I can certainly be convinced that parsing DerivedNames.txt is the right path here, and then manually-ish fill in missing data from UnicodeData.txt / NameAliases.txt.

Reply:
Our `display_name` is not the same as the Unicode name anyway, and intentionally so. One of the reasons is that the name `<control>` is useless, and those code points are completely missing in DerivedNames.txt. Another reason is the missing corrections: UnicodeData.txt and DerivedNames.txt contain the `BRAKCET` typo, so I assume all the other corrections are also missing.

No, I don't think DerivedNames.txt is useful for our usecase.

> would publish these formatting rules in a UCD file

We don't really rely on the formatting rules for several reasons. `display_name` is just a useful string for humans, and currently the only usecase is in FontEditor. So let's not try to follow the Unicode formatting rules too closely when they don't make sense (e.g. the name `<control>`, or wrong/typo'd names).

On the other hand, I don't really care whether we call it `CJK UNIFIED IDEOGRAPH-3400` or `CJK UNIFIED IDEOGRAPH EXTENSION A, 0x3400` or `CJK IDEOGRAPH EXTENSION A-3400` or something to that effect. Mostly I just saw a juge block of hardcoded data, and wondered why we don't use the data that's already in the downloaded file.

Reply:
In the case of getting rid of `<control>` (which the standard says is to be treated as a null string, so yes, useless) - I totally agree on not following the standard for helpful display names. Making corrections on e.g. `BRAKCET` is actually following the standard.

In the case of replacing `<Some name, First/Last>`, I do not agree with not following the standard. The standard formally defines the name between the angle brackets as just the range identifier, and is quite clear that implementations should derive (or parse DerviedNames.txt) the actual name.

I get not wanting hard-coded data, which is why I'd be willing to get these names from DerviedNames.txt to avoid hard-coding anything here. But the unfortunate truth with implementing Unicode is there's always going to be quite a bit of hard-coded data - not just our TR-44 implementation here, but also in our TR-35 implementation.

Reply:
Great, then we agree on the important part, and I don't feel strongly about the other stuff.

However, I do have a suggestion: What do we need the "actual name" for? Nothing, currently. The `display_name` is only informational in FontEditor, so I suggest we thrust as much information into the user's eyes as reasonably possible, so adding `Extension A` is nice, and the exact formatting is irrelevant.

Reply:
>  I suggest we thrust as much information into the user's eyes as reasonably possible, so adding Extension A is nice

Is it nice? What information does that provide you[^1]? More words doesn't mean better information.

I would want nothing other than correct, exactly Google-able names. I just searched ["CJK Ideograph Extension A-3400"](https://www.google.com/search?q=%22CJK+Ideograph+Extension+A+3400%22) (and several variations thereof), and I get 3 results, all of which look...oddly malware-ish (weird, tbh). If I search ["CJK UNIFIED IDEOGRAPH-3400"](https://www.google.com/search?q=%22CJK+UNIFIED+IDEOGRAPH-3400%22), the real name, I get great Unicode-related results, and Google even knows to inline a table with data about that code point.

[^1]: Extension X vs. Extension X+1 means nothing more than Extension X was added to the standard before X+1. The same information is known by the code point value.

Reply:
That's a good point. Again, I don't feel strongly about this, go ahead ^^